sdk/src/assertions/asset_reference.rs

```diff
@@ -0,0 +1,145 @@
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    assertion::{Assertion, AssertionBase, AssertionCbor},
+    assertions::labels,
+    error::Result,
+};
+
+const ASSERTION_CREATION_VERSION: usize = 2;
+
+/// An `AssetReference` assertion provides information on one or more locations of
+/// where a copy of the asset may be obtained.
+///
+/// This assertion contains a list of [`Reference`], each one declaring a location expressed as a URI and
+/// optionally a description. The URI may be either a single asset or it may reference a directory.
+///
+/// <https://spec.c2pa.org/specifications/specifications/2.2/specs/C2PA_Specification.html#_asset_reference>
+#[derive(Deserialize, Serialize, Debug, PartialEq)]
+pub struct AssetReference {
+    pub references: Vec<Reference>,
+}
+
+impl AssetReference {
+    pub const LABEL: &'static str = labels::ASSET_REFERENCE;
+
+    /// Creates an AssetReference to a location.
+    pub fn new(reference: Reference) -> Self {
```

Comment:
Given that the Reference is just a struct with a URL, I think we can just pass the Url as a String here and not bother with exporting the Reference type.